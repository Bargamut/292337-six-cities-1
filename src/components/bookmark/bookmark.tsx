import * as React from 'react';

import withFavorite from '../../hocs/with-favorite/with-favorite';

interface Props {
  isFavorite: boolean
  onFavoriteClick: () => void
  iconWidth?: number
  iconHeight?: number
  className?: string
  caption?: string
}

const Bookmark = (props: Props) => {
  const {isFavorite, onFavoriteClick, iconWidth, iconHeight, className, caption} = props;

  const blockClassName = className || `place-card`;
  const activeClass = isFavorite ? `${blockClassName}__bookmark-button--active` : ``;

  return (
    <button className={`${blockClassName}__bookmark-button ${activeClass} button`} type="button" onClick={onFavoriteClick}>
      <svg className={`place-card__bookmark-icon`} width={iconWidth || 18} height={iconHeight || 19}>
        <use xlinkHref="#icon-bookmark"></use>
      </svg>

      <span className="visually-hidden">{caption || `In bookmarks`}</span>
    </button>
  );
};

export {Bookmark};
export default withFavorite(Bookmark);
