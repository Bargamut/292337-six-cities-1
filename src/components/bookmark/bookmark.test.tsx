import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {Bookmark} from './bookmark';

it(`Bookmark correctly renders in NO favorite item`, () => {
  const header = renderer.create(
      <Bookmark
        isFavorite={false}
        onFavoriteClick={jest.fn()}
      />
  ).toJSON();

  expect(header).toMatchSnapshot();
});

it(`Bookmark correctly renders in favorite item`, () => {
  const header = renderer.create(
      <Bookmark
        isFavorite={true}
        onFavoriteClick={jest.fn()}
      />
  ).toJSON();

  expect(header).toMatchSnapshot();
});
