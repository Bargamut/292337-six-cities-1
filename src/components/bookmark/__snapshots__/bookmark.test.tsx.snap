// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark correctly renders in NO favorite item 1`] = `
<button
  className="place-card__bookmark-button  button"
  onClick={[MockFunction]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;

exports[`Bookmark correctly renders in favorite item 1`] = `
<button
  className="place-card__bookmark-button place-card__bookmark-button--active button"
  onClick={[MockFunction]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;
